testing/test_soft_limits.py

```diff
@@ -1,169 +1,170 @@
 
-    def test_ExceedMaxYLimit_SoftLimitsOff(self):
-        hardware.both_axes.current_position = (hardware.both_axes.y_soft_upper_limit - 1, 1)
-        line_segment = curves.LineSegment([hardware.both_axes.y_soft_upper_limit - 1, 1], [hardware.both_axes.y_soft_upper_limit + 10, 1])
+    def _follow_line_with_soft_limits_off_and_assert_no_warning(self, line_segment):
+        hardware.current_location = line_segment.start
         with warnings.catch_warnings(record=True) as w:
-            # Run test
-            with self.assertRaises(UnexpectedLimitSwitchError):
-                self._test_soft_limit(line_segment, False)
+            try:
+                hardware.both_axes.follow(line_segment, pen_speed=100, use_soft_limits=False, suppress_limit_warnings=False)
+            except limit_switches.UnexpectedLimitSwitchError:
+                pass
 
-            # Verify no warnings
             self.assertEqual(len(w), 0)
 
     @staticmethod
-    def _test_soft_limit(curve, use_limits=True, suppress_warnings=False):
-        temp_stdout = StringIO()
-        with contextlib.redirect_stdout(temp_stdout):
-            hardware.both_axes.follow(curve, pen_speed=100, use_soft_limits=use_limits, suppress_limit_warnings=suppress_warnings)
+    def _build_segment_crossing_min_x_soft_limit():
+        start_location = (hardware.y_axis.current_location, hardware.both_axes.x_soft_lower_limit + 1)
+        target_location = (hardware.y_axis.current_location, hardware.both_axes.x_soft_lower_limit - 1)
+        return curves.LineSegment(start_location, target_location)
+
+    @staticmethod
+    def _build_segment_crossing_max_x_soft_limit():
+        start_location = (hardware.y_axis.current_location, hardware.both_axes.x_soft_upper_limit - 1)
+        target_location = (hardware.y_axis.current_location, hardware.both_axes.x_soft_upper_limit + 1)
+        return curves.LineSegment(start_location, target_location)
+
+    @staticmethod
+    def _build_segment_crossing_min_y_soft_limit():
+        start_location = (hardware.both_axes.y_soft_lower_limit + 1, hardware.x_axis.current_location)
+        target_location = (hardware.both_axes.y_soft_lower_limit - 1, hardware.x_axis.current_location)
+        return curves.LineSegment(start_location, target_location)
+
+    @staticmethod
+    def _build_segment_crossing_max_y_soft_limit():
+        start_location = (hardware.both_axes.y_soft_upper_limit - 1, hardware.x_axis.current_location)
+        target_location = (hardware.both_axes.y_soft_upper_limit + 1, hardware.x_axis.current_location)
+        return curves.LineSegment(start_location, target_location)
+
+    # @staticmethod
+    # def _test_soft_limit(curve, use_limits=True, suppress_warnings=False):
+    #     temp_stdout = StringIO()
+    #     with contextlib.redirect_stdout(temp_stdout):
+    #         hardware.both_axes.follow(curve, pen_speed=100, use_soft_limits=use_limits, suppress_limit_warnings=suppress_warnings)
 
```

Comment:
Ooh sorry - pycharm had folded it up so I didn't see that!
It's not actually a test, but the common work you had previously factored out from the other tests.
However, the redirection of `stdout` isn't necessary since you changed from print statements to warnings. Therefore, I have just replaced its usage with `hardware.both_axes.follow(...)`.